pkg/secrethub/http.go

```diff
@@ -97,29 +97,18 @@ type httpClient struct {
 
 // newHTTPClient configures a new httpClient and overrides default values
 // when opts is not nil.
-func newHTTPClient(authenticator auth.Authenticator, opts *ClientOptions) *httpClient {
+func newHTTPClient() *httpClient {
```

Comment:
Oh wow that's actually pretty nasty yes. I feel because we split them up into two separate arguments, it should support it if one or the other is not configured/provided. 

Also there are a handful of API calls you can make without authentication, so I feel you should be able to create a client without authentication.

Reply:
Oke, so as you say above and as we have discussed in person at the office today, we'd like the authenticater and decrypter to be optional on the client constructor.

As the client has functions that don't require a decrypter and the http client has functions that don't require authentication, these can be omitted there as well (so are optional to configure).

For the functions that need them they are required. So an error should be returned when they are not present (in a first implementation this can depend on the server returning an unauthenticated response).

Reply:
#100